frontends/election-manager/src/components/export_ballot_pdfs_button.tsx

```diff
@@ -0,0 +1,375 @@
+import { Buffer } from 'buffer';
+import React, { useCallback, useContext, useState, useMemo } from 'react';
+import styled from 'styled-components';
+import { join } from 'path';
+import { BallotLocales } from '@votingworks/types';
+
+import {
+  assert,
+  usbstick,
+  throwIllegalValue,
+  BALLOT_PDFS_FOLDER,
+  BallotStyleData,
+} from '@votingworks/utils';
+import {
+  Modal,
+  Prose,
+  UsbControllerButton,
+  isAdminAuth,
+  isSuperadminAuth,
+} from '@votingworks/ui';
+import { LogEventId } from '@votingworks/logging';
+import {
+  getBallotArchiveFilename,
+  getBallotPath,
+  getBallotStylesData,
+  sortBallotStyleDataByPrecinct,
+} from '../utils/election';
+
+import { AppContext } from '../contexts/app_context';
+import { HandMarkedPaperBallot } from './hand_marked_paper_ballot';
+import { Button } from './button';
+import { LinkButton } from './link_button';
+import { Loading } from './loading';
+
+import { BallotMode } from '../config/types';
+import { DownloadableArchive } from '../utils/downloadable_archive';
+import { BallotTypeToggle } from './ballot_type_toggle';
+import { BallotModeToggle } from './ballot_mode_toggle';
+import { PrintableArea } from './printable_area';
+import { DEFAULT_LOCALE } from '../config/globals';
+import { generatePdfExportMetadataCsv } from '../utils/generate_pdf_export_metadata_csv';
+
+const { UsbDriveStatus } = usbstick;
+const UsbImage = styled.img`
+  margin-right: auto;
+  margin-left: auto;
+  height: 200px;
+`;
+
+const CenteredOptions = styled.div`
+  text-align: center;
+`;
+
+enum ModalState {
+  BEGIN_EXPORT = 'begin export',
+  GENERATE_FILES = 'generate files',
+  END_EXPORT = 'end export',
+  DONE = 'done',
+  ERROR = 'error',
+}
+
+const defaultBallotLocales: BallotLocales = { primary: DEFAULT_LOCALE };
+
+export function ExportBallotPdfsButton(): JSX.Element {
+  const { electionDefinition, usbDriveStatus, usbDriveEject, auth, logger } =
+    useContext(AppContext);
+  assert(electionDefinition);
+  assert(isAdminAuth(auth) || isSuperadminAuth(auth));
+  const userRole = auth.user.role;
+  const { election, electionHash } = electionDefinition;
+
+  const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+  const [modalState, setModalState] = useState(ModalState.BEGIN_EXPORT);
+  const [modalError, setModalError] = useState<Error | undefined>(undefined);
+  const [archive, setArchive] = useState(new DownloadableArchive());
+  const [ballotMode, setBallotMode] = useState(BallotMode.Official);
+  const [isAbsentee, setIsAbsentee] = useState(false);
+  const [ballotIndex, setBallotIndex] = useState(0);
+
+  const defaultArchiveFilename = useMemo<string>(
+    () => getBallotArchiveFilename(electionHash, ballotMode, isAbsentee),
+    [electionHash, ballotMode, isAbsentee]
+  );
```

Comment:
For simpler computations like the one in `getBallotArchiveFilename`, I lean toward not using `useMemo`. `useMemo` has overhead of its own that can end up negating performance benefits for simple computations, and it can also make code harder to read